generate/src/util.rs

```diff
@@ -60,6 +72,21 @@ pub fn http_get(url: &str) -> Result<String> {
 mod tests {
     use super::*;
 
+    #[test]
+    fn input_name_to_ident_empty() {
+        assert_quote!(expand_input_name(0, ""), { p0 });
+    }
+
+    #[test]
+    fn input_name_to_ident_keyword() {
+        assert_quote!(expand_input_name(0, "self"), { self_ });
+    }
+
+    #[test]
+    fn input_name_to_ident_snake_case() {
+        assert_quote!(expand_input_name(0, "CamelCase1"), { camel_case_1 });
+    }
```

Comment:
Why does self wind up with an underscore and not CamelCase1?

Reply:
`self` is a keyword. As a workaround, we append `_` to it so that there aren't compiler errors. From the doc comments:
```
/// Expands an identifier string into a token and appending `_` if the
/// identifier is for a reserved keyword.
///
/// Parsing keywords like `self` can fail, in this case we add an underscore.
```